### ISSUE TITLE
[BUG] Conflicts with |center image extension from ITS Theme which is no longer working

### ISSUE THREAD

**Issue by scarfur** [xRyul/obsidian-image-converter#125]
Hey there,

after updating to the [latest version](https://github.com/xRyul/obsidian-image-converter/releases/tag/1.2.17) of Image Converter, the `|center` image extension from [ITS Theme's snippets](https://github.com/SlRvb/Obsidian--ITS-Theme/tree/main/Snippets) (while using minimal theme) stopped working. Seems to affect Live Preview Mode only, tested in multiple Vaults.

**_Example:_**
`![[image.svg|hmicro center]]`
same unwanted result for:
`![[image.webp|center]]`

**Centered images get aligned to the left suddenly. When deactivating Image Converter, everything seems fine again?!**

Before I start searching for a fix, I quickly wanted to drop this here. If I find any more conflicts I would edit this post.

Greetings :)

**Comment by xRyul** [comment]
Hey, many thanks for reporting! 

I patched it just now, and while testing on ITS default Theme - the `center` positioning seems to work. But `LEFT` and `RIGHT` only works in Reading mode, is that's correct? I tried using the snippets you have linked, and those behave the same way. I tried disabling Image Converter plugin, but it seems to have no effect. 

ITS snippets:
Center - works in live/reading mode
Left/right - only shows up in reading mode

Regardless, now, you will be able to use any custom snippet as long as you do not set alignment via the plugin. 

When you set alignment via the plugin, plugins styling will come into effect and it will overwrite ITS snippet. For instance, if via the sippet you set it to `|center` and then via right-click you also set it to `right` - then precedence will be for the right alignment. Although, if you then disable `right` alignment via the right-click, it will then default back to using ITS snippet. Any resizing/alignment you do in Reading mode will show in Live preview, and vice-versa, any resizing/positioning you do in Live mode will show up in Reading mode too. 

Basically, plugin won't affect any snippet or custom CSS unless you explicitly tell it to do so.

Although won't be out for quite some time just yet. 
![Obsidian_cMRMk8UEHC](https://github.com/user-attachments/assets/7200147b-6c27-4188-ae2e-daac5a7e80a8)

**Comment by scarfur** [comment]
Many thanks for you detailed answer.

Strangely I did not set any alignment via Image Converter. Furthermore I noticed, on mobile there are some miss-alignments differing from the desktop version. For example one of the same two test images, both with a center alignment (snippet) was displayed left and one correctly was centered. I do not get it.
For me, there definitely was a difference when deactivating the plugin and still there is.

I am going to test it again, maybe I find a clue what could be the reason for this behavior - hopefully.

**Comment by scarfur** [comment]
I created a new test vault with ITS Theme and Image Converter only and can reproduce it like this:

1. did not touched any settings
2. insert new image
3. add `|center`
4. swap between view modes (correctly centered in live preview, wrong alignment in reading mode)
5. **deactivate plugin**
6. **see correct alignment in all view modes**

**Comment by xRyul** [comment]
> Many thanks for you detailed answer.
> 
> Strangely I did not set any alignment via Image Converter. Furthermore I noticed, on mobile there are some miss-alignments differing from the desktop version. For example one of the same two test images, both with a center alignment (snippet) was displayed left and one correctly was centered. I do not get it. For me, there definitely was a difference when deactivating the plugin and still there is.
> 
> I am going to test it again, maybe I find a clue what could be the reason for this behavior - hopefully.

Sorry, I had to be more clear. I patched it, but have not released it yet. The demo video I have included is just to show that ITS theme and ITS snippets now work too.

I will update you when it comes out 🙏

**Comment by scarfur** [comment]
Sorry, I had misunderstood that. Thank you for the clarification and your constant efforts with the plugin and the submitted issues. Have a great festive season! ⛄

**Comment by elvarb** [comment]
I'm using a custom css snippet to align images in the same way that the ITS theme does.

```
img[alt*="center"] {
    display: block;
    margin-left: auto;
    margin-right: auto;
    }
```

Currently with the plugin enabled the image is always left aligned and the snippet is overruled., using the right click menu to align the image does work very well.

To get around this I just modified the snippet as follows and now it works.
```

img[alt*="center"] {
    display: block !important;
    margin-left: auto;
    margin-right: auto;
    }
```

Now with the plugin itself, where are the alignment information stored if not within the image embed link? 

Obsidian supports resizing images in the embed link, so would it not make sense to have the plugin store the image alignment there as well?

**Comment by xRyul** [comment]
`center`, `hmicro center` should now work in 1.2.24

![image](https://github.com/user-attachments/assets/c8ce5d20-af52-45f0-aabf-3ba4a396ca3a)

If you encounter any issues, please comment back.

**Comment by elvarb** [comment]
Seems that this was broken again with the caption feature. Now with images defined |center|200 for example show center as the caption.

![Image](https://github.com/user-attachments/assets/d98621fc-9ad3-44c6-8615-615f0dca5b80)
